### PR TITLE
DX: migrate to hardhat

### PR DESCRIPTION
**Pull request by 0xGabi**
Changes to migrate from builder to hardhat. I tested all features and everything is working:

- [x] hardhat compile
- [x] hardhat test
- [x] hardhat coverage
- [x] hardhat publish (using the new plugin [`@1hvie/hardhat-aragon`](https://github.com/1Hive/hardhat-aragon))

The only caveat is that a few tests are failing. All with the same error:
```bash
Error: Transaction reverted without a reason string
      at <UnrecognizedContract>.<unknown> (0x75a3e005ab49bc079d547fb2d1b5857e7aec7a39)
      at <UnrecognizedContract>.<unknown> (0x8b7644e22b5e3abcfb4c34c7e04788ce72f81fc3)
      at <UnrecognizedContract>.<unknown> (0x8b7644e22b5e3abcfb4c34c7e04788ce72f81fc3)
      at TruffleContract.makeBuyOrder (node_modules/@nomiclabs/truffle-contract/lib/execute.js:169:26)
      at makeBuyOrder (test/augmented-bonding-curve.js:259:30)
      at Context.<anonymous> (test/augmented-bonding-curve.js:1176:21)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)

```

I believe it has to do with one of the assertions functions not handling the errors correctly. Similar to what I described in https://github.com/CommonsSwarm/augmented-bonding-curve/pull/21#discussion_r739590405 for the `assertThrow`.

Note that because we are using [`@1hvie/hardhat-aragon`](https://github.com/1Hive/hardhat-aragon) plugin the `arapp.json` file is deprecated. I was not sure if we should delete it or keep it around. What do you think?

---
TODO:

- [x] Debug why a few tests keep failing (please if you can help with this one that would be awesome 🙌)

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# network specific node uri : `"ETH_NODE_URI_" + networkName.toUpperCase()`
```
Comment (0xGabi) [inline .env.example:1]: The intention of this file is to have an easy way to keep ENV variables accessible (e.g. mnemonic, nodes, secrets)

File: .gitattributes [hunk 2]
```diff
@@ -1 +0,0 @@
-*.sol linguist-language=Solidity
```
Comment (0xGabi) [inline .gitattributes:1]: This file is no longer necessary as GitHub had support for `.sol` files.

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (0xGabi) [inline .vscode/launch.json:1]: I keep this file as a reference on how to attach to the hardhat process and debug using vscode.

File: hardhat.config.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 10000,
+          },
+        },
+      },
+    ],
+  },
+  aragon: {
```
Comment (0xGabi) [inline hardhat.config.js:28]: The `aragon` config extension replaces the `arapp.json` file. To have more context check README of [`@1hive/hardhat-aragon` plugin](https://github.com/1Hive/hardhat-aragon/blob/master/README.md#config-extensions).

File: hardhat.config.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            blockNumber: process.env.HARDHAT_FORK_NUMBER
+              ? parseInt(process.env.HARDHAT_FORK_NUMBER)
+              : undefined,
+          }
+        : undefined,
+    },
+    localhost: {
+      url: node_url('localhost'),
+      accounts: accounts(),
+      ensRegistry: '0xaafca6b0c89521752e559650206d7c925fd0e530',
```
Comment (0xGabi) [inline hardhat.config.js:97]: The `ensRegistry` is also an extension configured by `hardhat-aragon` plugin. That allows us to set up a custom `ensRegistry` that we will be using when publishing our app to AragonPM.

File: hardhat.config.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    },
+    mainnet: {
+      url: node_url('mainnet'),
+      accounts: accounts('mainnet'),
+      appEnsName: 'augmented-bonding-curve.aragonpm.eth',
```
Comment (0xGabi) [inline hardhat.config.js:102]: We are also able to configure an `appEnsName` per network if needed.

File: hardhat.config.js [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+      gas: 0,
+    },
+  },
+  ipfs: {
+    gateway: 'https://ipfs.blossom.software/',
+    pinata: {
+      key: process.env.PINATA_KEY || '',
+      secret: process.env.PINATA_SECRET_KEY || '',
+    },
+  },
```
Comment (0xGabi) [inline hardhat.config.js:147]: `ipfs` config extension gives us control over the IPFS connection that `hardhat-aragon` setup for us. Check the [README of the plugin](https://github.com/1Hive/hardhat-aragon/blob/master/README.md) for more details.

File: test/helpers/assertThrow.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
+const { assert } = require('chai')
```
Comment (0xGabi) [inline test/helpers/assertThrow.js:1]: We have to include a custom `assertThrow` file to handle a new error message format that hardhat returns and `@aragon/contract-helpers-test` do not handle it.

**Comment by 0xGabi** [comment]
@sembrestels we had two references of `MinimeToken.sol`. One from the `app-shared-minime` package and the other from the `minime` package. I remove one of them to prevent further issues.

**Comment by BradleyC** [comment]
ty ty it's so hard to type "builder" the wrong way I mistake everytiem

**Comment by sembrestels** [comment]
I rebased the commit and everything works as expected. Thanks @0xGabi!

**Pull request merged**